lib/list.js

```diff
@@ -64,7 +64,12 @@ function List(items, itemType, parent) {
   }
 
   List.prototype.toItem = function(item) {
-    return isClass(this.itemType) ? new this.itemType(item) : this.itemType(item);
+    if (isClass(this.itemType)) {
+      return new this.itemType(item);
+    } else {
+      if (Array.isArray(item)) return item;
```

Comment:
Do we want to test if `item instanceof List` to convert `List`, which is a special array?

Reply:
In that case it is breaking some test cases. The data structure which elicited the bug is a very unlikely one. This PR takes care of that scenario without side effects.

I think we should land the PR and move on.

Reply:
ok